### PR TITLE
Add deploy started and approval requested dates to dnenvironment

### PR DESCRIPTION
**Pull request by stojg**
This changeset also includes some cleanup on a couple of JS files and ensuring that an active deployment shows up in the "Upcoming deployments"  list.

### CODE REVIEW

File: code/model/DNEnvironment.php [hunk 1]
```diff
@@ -615,9 +615,18 @@ public function CurrentBuild() {
...
 	 * A list of past deployments.
-	 * @return ArrayList
+	 * @param string $orderBy - the name of a DB column to sort in descending order
+	 * @return \ArrayList
 	 */
-	public function DeployHistory() {
+	public function DeployHistory($orderBy = '') {
+
+		$sort = [];
+		if($orderBy != '') {
```
Comment (halkyon) [inline code/model/DNEnvironment.php]: Can we stick with the `if ($expression) {` syntax? The whitespace after the `if`, `foreach` or whatever?

Comment (halkyon) [inline code/model/DNEnvironment.php]: And remove that whitespace at the start of the DeployHistory function? :D

File: code/control/DNDeploymentHandlers.php [hunk 2]
```diff
@@ -5,15 +5,20 @@
 class DNDeploymentHandlers extends Object {
 
 	public function onSubmit(TransitionEvent $e) {
+		/** @var DNDeployment $deployment */
 		$deployment = $e->getStateMachine()->getObject();
+		$deployment->DeployRequested = SS_Datetime::now()->Rfc2822();
```
Comment (halkyon) [inline code/control/DNDeploymentHandlers.php]: "Deploy date" in the history - is it the start, or the _end_ of the deployment job? :)

File: code/model/DNEnvironment.php [hunk 3]
```diff
@@ -615,9 +615,18 @@ public function CurrentBuild() {
...
+			$sort[$orderBy] = 'DESC';
+		}
+		// default / fallback sort order
+		$sort['LastEdited'] = 'DESC';
```
Comment (halkyon) [inline code/model/DNEnvironment.php]: This is not effective, you are not passing `$sha` into the ORM call underneath :)

File: code/control/DNDeploymentHandlers.php (hunk 2 above)
Comment (stojg) [inline code/control/DNDeploymentHandlers.php]: I have no clue, I went with the start date.

File: code/model/DNEnvironment.php (hunk 3 above)
Comment (stojg) [inline code/model/DNEnvironment.php]: LOL! thanks.

File: code/control/DNDeploymentHandlers.php (hunk 2 above)
Comment (halkyon) [inline code/control/DNDeploymentHandlers.php]: Fair enough. I think we'll stick with that for now then.

**Pull request merged**